local-volume/provisioner/pkg/common/common.go

```diff
@@ -169,26 +169,25 @@ func GetDefaultVolumeConfig() map[string]MountConfig {
 // VolumeConfigToConfigMapData converts volume config to configmap data.
 func VolumeConfigToConfigMapData(config map[string]MountConfig) (map[string]string, error) {
 	configMapData := make(map[string]string)
-	for class, data := range config {
-		var val []byte
-		var err error
-		if val, err = json.Marshal(data); err != nil {
-			return nil, fmt.Errorf("unable to unmarshal config for class %v: %v", class, err)
-		}
-		configMapData[class] = string(val)
+	val, err := yaml.Marshal(config)
+	if err != nil {
+		return nil, fmt.Errorf("unable to Marshal volume config: %v", err)
 	}
+	configMapData["storageClassMap"] = string(val)
```

Comment:
`GetVolumeConfigFromMountedConfigMap()` should be modified too right?

Reply:
Do not think so, as at the end of GetVolumeConfigFromMountedConfigMap it calls new func         
return ConfigMapDataToVolumeConfig(storageClassMapping)

Reply:
The function used to assume that each storageclass name was its own file.  But now there's only one file for all storage class mappings.  So I think it does need to be updated.

Reply:
Ok, I will rerun tests and let you know.

Reply:
```
I0911 21:32:13.488179       1 main.go:79] Running provisioner with config map[local-fast:{HostDir:/mnt/ssds MountDir:/local-ssds} local-slow:{HostDir:/mnt/hdds MountDir:/local-hdds} local-storage:{HostDir:/mnt/disks MountDir:/mnt/local-storage/mnt~disks}]
I0911 21:32:13.488223       1 controller.go:41] Initializing volume cache
I0911 21:32:13.488367       1 populator.go:85] Starting Informer controller
I0911 21:32:13.488391       1 populator.go:89] Waiting for Informer initial sync
I0911 21:32:14.488679       1 controller.go:69] Controller started
I0911 21:32:14.489034       1 discovery.go:161] Found new volume of volumeType "file" at
host path "/mnt/disks/vol1" with capacity 51973496832, creating Local PV "local-pv-8f263585"
I0911 21:32:14.497302       1 cache.go:55] Added pv "local-pv-8f263585" to cache
I0911 21:32:14.497517       1 discovery.go:178] Created PV "local-pv-8f263585" for volume at "/mnt/disks/vol1"
I0911 21:32:14.497552       1 discovery.go:161] Found new volume of volumeType "file" at host path "/mnt/disks/vol2" with capacity 55346163712, creating Local PV "local-pv-1fb30a1c"
I0911 21:32:14.499424       1 cache.go:55] Added pv "local-pv-1fb30a1c" to cache
I0911 21:32:14.500107       1 discovery.go:178] Created PV "local-pv-1fb30a1c" for volume at "/mnt/disks/vol2"
I0911 21:32:14.500845       1 cache.go:64] Updated pv "local-pv-8f263585" to cache
I0911 21:32:14.502696       1 cache.go:64] Updated pv "local-pv-1fb30a1c" to cache
```
Here is the provisioner log with current code, looks good to me.

Reply:
It's works now because storageClassMap is the only parameter we support.  However, if we add more parameters in the future, `GetVolumeConfigFromMountedConfigMap` won't work anymore.